### PR TITLE
added homepage navbar footer

### PR DESCRIPTION
**Pull request by Hnh013**
added components : Navigation bar , Footer , Category , Featured
added Pages : Home
added routing : one route for home page

### CODE REVIEW

**Review by hrishi82 — APPROVED** [review]
Code looks good, can be merged

File: src/frontend/templates/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    const [navSearchClass, setSearchClass] = useState('nav-g-search');
+
+    const toggleNavBar = (e) => {
+        navlinksClass === 'nav-links'
+          ? setNavlinksClass('nav-links-responsive')
+          : setNavlinksClass('nav-links');
+        navSearchClass === 'nav-g-search responsive'
+          ? setSearchClass('nav-g-search')
+          : setSearchClass('nav-g-search responsive');
+    };
```
Comment (tejas-muthya37) [inline src/frontend/templates/components/Navbar.js:17]: I think this snippet can be simplified in the following way.

<element className={"common-class" + (stateVariable ? "conditional-class" : "")}

File: src/frontend/templates/pages/Home.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    return (
+        <main className='main-home'>
+            <section>
+                <div>
+                    <span className='f-w-800 f-2x txt-sun d-flex jc-c py-sm'>Shop By Category</span>
+                </div>
+                <div className='d-flex fw-wrap ai-c jc-sa gap-1 py-sm px-sm'>
+                    {categories.map((categoryObj,index) => 
+                    <Category key={index} categoryObj={categoryObj} /> )}
```
Comment (tejas-muthya37) [inline src/frontend/templates/pages/Home.js:21]: The db has IDs for categories that can be used as the key here.

File: src/frontend/templates/pages/Home.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    className='btn btn-calm bdr-rad-sm'>Shop Now</button>
+                </div>
+            </section>
+            <section>
+                <div>
+                    <span className='f-w-800 f-2x txt-fire d-flex jc-c py-sm'>Featured Products</span>
+                </div>
+                <div className="category-row d-flex ai-c mx-auto px-sm py-sm">   
+                    <div className="d-flex ai-c">
+                        <button onClick={() => scroll(-200)} className="material-icons arrow-left btn-fab btn-over">
```
Comment (tejas-muthya37) [inline src/frontend/templates/pages/Home.js:40]: Have you considered if the -200 here will work as intended on screens of different widths?

**Review by tejas-muthya37 — APPROVED** [review]
Really liked the quality of your code. The naming conventions that you've used for the class names of the elements are a little different to what I've seen, they may be a convention that I'm aware of. If that's the case, they look very consistent. If not, you can try to make them a little more descriptive.

File: src/frontend/templates/components/Navbar.js (hunk 1 above)
Comment (Hnh013) [reply src/frontend/templates/components/Navbar.js:17]: hi tejas , thanks for the suggestion , will implement this in future iteration !

File: src/frontend/templates/pages/Home.js (hunk 3 above)
Comment (Hnh013) [reply src/frontend/templates/pages/Home.js:40]: hi tejas, yes i tested these on mobile phone ( ~400px  width ) they worked

**Pull request merged**